x-pack/plugins/security_solution/public/detections/pages/alert_details/index.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useEffect } from 'react';
+import { Switch, useParams } from 'react-router-dom';
+import { Route } from '@kbn/kibana-react-plugin/public';
+import { ALERT_RULE_NAME, TIMESTAMP } from '@kbn/rule-data-utils';
+import { EuiSpacer } from '@elastic/eui';
+import { useDispatch } from 'react-redux';
+import { timelineActions } from '../../../timelines/store/timeline';
+import { TimelineId } from '../../../../common/types';
+import { useGetFieldsData } from '../../../common/hooks/use_get_fields_data';
+import { useSourcererDataView } from '../../../common/containers/sourcerer';
+import { SourcererScopeName } from '../../../common/store/sourcerer/model';
+import { useSpaceId } from '../../../common/hooks/use_space_id';
+import { SpyRoute } from '../../../common/utils/route/spy_routes';
+import { getAlertDetailsTabUrl } from '../../../common/components/link_to';
+import { AlertDetailRouteType } from './types';
+import { SecuritySolutionTabNavigation } from '../../../common/components/navigation';
+import { getAlertDetailsNavTabs } from './utils/navigation';
+import { DEFAULT_ALERTS_INDEX, SecurityPageName } from '../../../../common/constants';
+import { eventID } from '../../../../common/endpoint/models/event';
+import { useTimelineEventsDetails } from '../../../timelines/containers/details';
+import { AlertDetailsLoadingPage } from './components/loading_page';
+import { AlertDetailsErrorPage } from './components/error_page';
+import { AlertDetailsHeader } from './components/header';
+import { DetailsSummaryTab } from './tabs/summary';
+
+export const AlertDetailsPage = memo(() => {
+  const { detailName: eventId } = useParams<{ detailName: string }>();
+  const dispatch = useDispatch();
+  const currentSpaceId = useSpaceId();
+  const sourcererDataView = useSourcererDataView(SourcererScopeName.detections);
+  const spaceAlertsIndexAlias = `${DEFAULT_ALERTS_INDEX}-${currentSpaceId}`;
+  const [loading, detailsData, searchHit, dataAsNestedObject] = useTimelineEventsDetails({
+    indexName: spaceAlertsIndexAlias,
+    eventId,
+    runtimeMappings: sourcererDataView.runtimeMappings,
+    skip: !eventID,
+  });
+  const dataNotFound = !loading && !detailsData;
+  const hasData = !loading && detailsData;
+
+  // Example of using useGetFieldsData. Only place it is used currently
+  const getFieldsData = useGetFieldsData(searchHit?.fields);
+  const timestamp = getFieldsData(TIMESTAMP) as string;
+  const ruleName = getFieldsData(ALERT_RULE_NAME) as string;
+
+  useEffect(() => {
+    // TODO: move detail panel to it's own redux state
```

Comment:
Work has been done to split up the reducer state for the tables and timeline, but pages like cases and now alert details maintain timeline reducer state because of the `expandedDetail` which shows the alert_detail, host_detail, network_detail, and user_detail flyouts. There should be separate state for that. It might even be able to be removed from redux entirely 🤔

Reply:
What is the benefit for moving it out of redux? The type of 
```
export type TimelineExpandedDetailType =
  | TimelineExpandedEventType
  | TimelineExpandedHostType
  | TimelineExpandedNetworkType
  | TimelineExpandedUserType;
  ```
  seems fine to me, I think where state related things both trip up typescript and make things seem more confusing than they should be is due to the existence of x-pack/plugins/security_solution/public/timelines/containers/active_timeline_context.ts . Delete that, and selectors in a shared hook could be more useful probably.

Reply:
I'm fine leaving it in Redux. I just don't think it needs to be explicitly tied with the `timeline` or `table` slice of state. The only reason being, both the cases page and now this details page have no need for anything else in the timeline or table redux state, yet both need to createTimeline to get the flyout to work. I'm fine using another slice of redux, but the cases team doesn't use redux so it would be cool to have a state management agnostic solution for the flyout. And yea, I agree that with that we should also delete ` x-pack/plugins/security_solution/public/timelines/containers/active_timeline_context.ts`... it's a lot of over-engineering.